backend/src/main/java/gov/cdc/usds/simplereport/api/queue/QueueMutationResolver.java

```diff
@@ -24,25 +25,47 @@
 
   private final TestOrderService _tos;
   private final PersonService _ps;
+  private final DeviceTypeService _dts;
 
-  public QueueMutationResolver(TestOrderService tos, PersonService ps) {
+  public QueueMutationResolver(TestOrderService tos, PersonService ps, DeviceTypeService dts) {
     _tos = tos;
     _ps = ps;
+    _dts = dts;
   }
 
   public AddTestResultResponse addTestResultNew(
-      String deviceID, String result, UUID patientID, Date dateTested) throws NumberParseException {
-    return _tos.addTestResult(deviceID, TestResult.valueOf(result), patientID, dateTested);
+      String deviceID, UUID deviceSpecimenType, String result, UUID patientID, Date dateTested)
+      throws NumberParseException {
+    UUID deviceSpecimenTypeId =
+        deviceSpecimenType == null
+            ? _dts.getDefaultForDeviceId(UUID.fromString(deviceID)).getInternalId()
+            : deviceSpecimenType;
+
+    return _tos.addTestResult(
+        deviceSpecimenTypeId, TestResult.valueOf(result), patientID, dateTested);
   }
 
-  public ApiTestOrder addTestResult(String deviceID, String result, UUID patientID, Date dateTested)
+  public ApiTestOrder addTestResult(
```

Comment:
We are indeed ready to remove `addTestResultNew`! It will be done in [#1912](https://app.zenhub.com/workspaces/simplereport-better-data-team-605b43d51ea9f700119a48ee/issues/cdcgov/prime-simplereport/1912), along with some associated cleanup.